README.md

```diff
@@ -76,45 +223,59 @@ You can also right-click on any element in the ARM Template Outline view to init
   - Sort template and template sections alphabetically
 
-## Extension configurations
+## Extension configuration
 
 You may be interested in adjusting the following extension configurations. These can be configured in [VS Code User Settings](https://code.visualstudio.com/docs/getstarted/settings).
 
-**Auto-detect ARM Templates**
+### Auto-detect ARM templates
 
 Enables auto-detection of deployment template files with the extension *.json or *.jsonc. If set to true (default), the editor language will automatically be set to Azure Resource Manager Template for any *.json/*.jsonc file which contains an appropriate Azure Resource Manager Template schema.
 
 ```
 "azureResourceManagerTools.autoDetectJsonTemplates": true,
 ```
 
-**Detect latest root schema**
+### Detect latest root schema
 
 Check if the root schema for deployment templates is using an out-of-date version and suggest updating it to the latest version.
 
 ```
 "azureResourceManagerTools.checkForLatestSchema": true,
 ```
 
-**Parameter files**
+### Parameter files
 
-Check if an opened template file has a matching params file and prompt to create a mapping.
+Check if an opened template file has a matching params file and prompt to create an association.
 
 ```
 "azureResourceManagerTools.checkForMatchingParameterFiles": true,
 ```
 
-Parameter file mappings are stored in the `azureResourceManagerTools.parameterFiles` user setting.
+Parameter file assoications are stored in the `azureResourceManagerTools.parameterFiles` user setting.
 
 ```
 "azureResourceManagerTools.parameterFiles": {}
 ```
 
+### Code lense
+
+Enable or disable all code lens features.
+
+```
+"azureResourceManagerTools.codelens.enable": true,
+```
+
+Enable code lens for parameter definitions and values.
+
+```
+ "azureResourceManagerTools.codelens.parameters": true,
+```
+
 ## Automatic Detection of deployment template files
 
 By default, the extension recognizes a .json or .jsonc file as a deployment template file based on the \$schema specified in the file (for instance, `https://schema.management.azure.com/schemas/2018-05-01/deploymentTemplate.json#`) and will switch the editor language to "Azure Resource Manager Template" automatically. If you do not want that behavior, you can set the `azureResourceManagerTools.autoDetectJsonTemplates` setting to false and use the below methods to determine which files to treat as deployment templates.
 
-Besides automatic detection, you can also use the `files.associations` setting to set up your own specific mappings based on specific files paths or patterns to mark them as deployment templates, e.g.
+Besides automatic detection, you can also use the `files.associations` setting to set up your own specific associations based on specific files paths or patterns to mark them as deployment templates, e.g.
 
 ```json
 "files.associations": {
```

Comment:
Need to update the troubleshooting section unrecognized api version so that it contains the actual error message (so people can search for it).

Reply:
Fixed